### DISCUSSION TITLE
Setting for air condition (AC)? Pulse with auto off but not return to the original status in HomeKit

### DISCUSSION THREAD

**Discussion by AlvaroLopezCarrillo** [mongoose-os-apps/shelly-homekit#1581] [Q&A]
Hello. I have doubts when configuring shelly 1 to simulate a press of the ON/OFF button of the thermostat of my wall air conditioner.

I currently have a shelly 1 but I did not program it and I need to change it because it is dead.

I need to simulate a press (1 second) BUT I want that in homekit the new state does not go back with the auto-off after 1 second, because the air conditioner will be in that new status.

Is this possible? Thanks!

**Comment by markirb** [comment]
No. Not in switch mode.

And maybe a little bit:

If you configure this as garage door you can have and configure the pulses. But you will not have state. 
The problem is that you have no way of keeping/syncing state if there is no feedback. 

So while theoretically this could be added a feature it will not really make sense. If you are away from home and your device incorrectly reboots/or somebody uses another switch you have no way of knowing the current state?

**Reply by AlvaroLopezCarrillo** [reply]
Hi,

I currently have it like this, I don't know how the person who sold me this shelly programmed it but I don't have his contact and the shelly doesn't work after 3 or 4 years.

I only want 1 pulse to simulate a button press (for example a switch that returns to off in 1 second) but I want the status not to change to off in the home app.

If the light goes out one day in the house it doesn't matter, I manually set the status in the home app.

So this is not currently possible?

Thank you very much!

**Comment by AlvaroLopezCarrillo** [comment]
Hello again. 

I'm sorry to answer again but do you think an update with this would be possible? 

Maybe something as easy as a check so that the auto-off after 1 second does not change the status to off?

This would be valid for any device that turns on with a press of 1 second and you want to maintain the status in the home app. 

Thank you!

**Comment by AlvaroLopezCarrillo** [comment]
@markirb 

Could this "no 2 status changes" be added when the "auto-off" is set in 1 second to simulate a pulse? 

Yesterday I left the house and my air conditioning was working for 12 hours without anyone at home, if in the pulse the status did not return to "off" I would have seen it in the "home" app and I would have turned it off.

I think it's a very interesting utility or directly add a "pulse" mode that only acts for 1 second and doesn't change the state 2 times. Could it be considered?

I appreciate it. 

Thank you!
